src/inmanta/ast/type.py

```diff
@@ -161,6 +200,43 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.element_type.is_primitive()
+
+    def corresponds_to(self, pytype: type[object]) -> bool:
+        if not typing_inspect.is_optional_type(pytype):
+            return False
+        # remove options from union
+        other_types = [tp for tp in typing_inspect.get_args(pytype) if not typing_inspect.is_optional_type(tp)]
+        # Typing union conveniently returns the base type if that is the only element
+        return self.element_type.corresponds_to(typing.Union[*other_types])
+
+    def as_python_type_string(self) -> "str | None":
+        return f"{self.element_type.as_python_type_string()} | None"
+
+    def to_python(self, instance: object) -> "object":
+        if isinstance(instance, NoneValue):
+            return None
+        return self.element_type.to_python(instance)
+
+    def has_custom_to_python(self) -> bool:
+        return self.element_type.has_custom_to_python()
+
+
+class Any(Type):
+
+    def corresponds_to(self, pytype: type[object]) -> bool:
+        return True
+
+    def as_python_type_string(self) -> "str | None":
+        return "object"
```

Comment:
This isn't right, is it?

Reply:
is it not?  I think object is the most precise representation of our 'Any'

OTOH, it can't be written in the model, so this code is unreachable. I can raise an exception here as well?

what do you prefer?

Reply:
On second thought, I think I agree with you that it matches.